New blog post: sail operator GA

fjglira opened the pull request:
## Description

Now that Sail Operator version 1.0.0 is out, we want to raise awareness in the community about the released version

## Reviewers

<!-- To help us figure out who should review this PR, please 
     put an X in all the areas that this PR affects. -->

- [ ] Ambient
- [X] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure
- [ ] Localization/Translation

<!-- If this is a localization PR, please replace this line with the URL of the original English document. -->

istio-testing commented:
Hi @fjglira. Thanks for your PR.

I'm waiting for a [istio](https://github.com/orgs/istio/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/istio/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=istio%2Fistio.io).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository.
</details>

Arhell reviewed (commented):
/ok-to-test

nrfox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,302 @@
+- Two update strategies are supported: `InPlace` and `RevisionBased`. Check our documentation for more information about the update types supported.
+- Support for multicluster Istio [deployment models](/docs/setup/install/multicluster/): multi-primary, primary-remote, external control plane. More information and examples in our [documentation](https://github.com/istio-ecosystem/sail-operator/blob/main/docs/README.md#multi-cluster).
+- Support for [multiple mesh](/docs/ops/deployment/deployment-models/#multiple-meshes): The Sail Operator supports running multiple meshes on a single cluster and associating each workload with a specific mesh. Each mesh is managed by a separate control plane. More information and examples can be found [here](https://github.com/istio-ecosystem/sail-operator/blob/main/docs/README.md#multiple-meshe ...
+- [Dual-Stack](https://github.com/istio-ecosystem/sail-operator/blob/main/docs/README.md#dual-stack-support) support since istio `v1.23` version.
```
Again I don't think this is really a Sail Operator feature since all it is doing is passing through the helm values.

nrfox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,302 @@
+## Try it out
+
+Do you want to try our operator?
+Going over this example will show you how to safely do an update of your istio control plane by using the revision-based upgrade strategy, this means you will have two Istio control planes running at the same time,allowing you to migrate workloads easily, minimizing the risk of traffic disruptions.
```
```suggestion
Going over this example will show you how to safely do an update of your istio control plane by using the revision-based upgrade strategy, this means you will have two Istio control planes running at the same time, allowing you to migrate workloads easily, minimizing the risk of traffic disruptions.
```

nrfox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,302 @@
+TEST SUITE: None
+{{< /text >}}
+
+### Check the operator pod running
```
```suggestion
### Ensure the operator pod is running
```

nrfox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,302 @@
+EOF
+{{< /text >}}
+
+Note that `IstioRevisionTag` has target reference to `Istio` resources with name default.
```
```suggestion
Note that the `IstioRevisionTag` has a target reference to the `Istio` resource with the name `default`.
```

nrfox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,302 @@
+default   NotReferencedByAnything   False    default-v1-24-2   4m43s
+{{< /text >}}
+
+Note that the `IstioRevisionTag` status is `NotReferencedByAnything`, this is because there is no resource referenced using the revision `default-v1-24-2`.
```
```suggestion
Note that the `IstioRevisionTag` status is `NotReferencedByAnything`, this is because there are currently no resources using the revision `default-v1-24-2`.
```

nrfox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,302 @@
+$ kubectl label namespace sample istio-injection=enabled
+{{< /text >}}
+
+As a result of the label of the namespace you will see that the `IstioRevisionTag` resource status will change to In Use true, this is because there is already one resource making reference to the revision `default-v1-24-2`.
```
```suggestion
After labeling the namespace you will see that the `IstioRevisionTag` resource status will change to In Use true, this is because there is now a resource using the revision `default-v1-24-2`.
```

nrfox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,302 @@
+sleep-6f87fcf556-k9nh9.sample     Kubernetes     SYNCED (29s)     SYNCED (29s)     SYNCED (29s)     SYNCED (29s)     IGNORED     istiod-default-v1-24-3-68df97dfbb-v7ndm     1.24.3
+{{< /text >}}
+
+- When an `IstioRevision` is no longer in use and is not the active revision of an `Istio` resource (for example not the version that is set in the `spec.version` field), the Sail Operator will delete it after a grace period, which defaults to 30 seconds. Confirm the deletion of the old control plane and `IstioRevision`:
```
```suggestion
- When an `IstioRevision` is no longer in use and is not the active revision of an `Istio` resource (for example, when it is not the version that is set in the `spec.version` field), the Sail Operator will delete it after a grace period, which defaults to 30 seconds. Confirm the deletion of the old control plane and `IstioRevision`:
```

nrfox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,302 @@
+
+## Conclusion
+
+The Sail Operator automates manual tasks, ensuring a consistent, reliable, and uncomplicated experience from initial installation to ongoing maintenance and upgrades of Istio versions in your cluster. The Sail Operator is a project part of the istio-ecosystem organization, and we encourage you to try it out and provide feedback to help us improve it, you can check our contribution guide [here](ht ...
```
```suggestion
The Sail Operator automates manual tasks, ensuring a consistent, reliable, and uncomplicated experience from initial installation to ongoing maintenance and upgrades of Istio in your cluster. The Sail Operator is an istio-ecosystem project, and we encourage you to try it out and provide feedback to help us improve it, you can check our contribution guide [here](https://github.com/istio-ecosystem/sail-operator/blob/main/CONTRIBUTING.md) for more information about how to contribute to the project.
```

nrfox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,302 @@
+Note that if you are migrating from the [deprecated and deleted] In-Cluster Istio operator, you can check this section in our [documentation](https://github.com/istio-ecosystem/sail-operator/tree/main/docs#migrating-from-istio-in-cluster-operator) where we explain the equivalence of resources, or you can try also our [resource converter](https://github.com/istio-ecosystem/sail-operator/tree/main/ ...
+
+## Main Features and support
+
```
What's not mentioned in this list is managing the different components of your control plane as a Kubernetes resource. There's a separate CRD for `Istio`, `IstioCNI`, `ZTunnel`, etc. and these pieces can each be managed and upgraded through their respective CRs.

fjglira replied:
Added a bullet point for this information

fjglira commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,302 @@
+- Support for multiple Istio versions.
+- Two update strategies are supported: `InPlace` and `RevisionBased`. Check our documentation for more information about the update types supported.
+- Support for multicluster Istio [deployment models](/docs/setup/install/multicluster/): multi-primary, primary-remote, external control plane. More information and examples in our [documentation](https://github.com/istio-ecosystem/sail-operator/blob/main/docs/README.md#multi-cluster).
+- Support for [multiple mesh](/docs/ops/deployment/deployment-models/#multiple-meshes): The Sail Operator supports running multiple meshes on a single cluster and associating each workload with a specific mesh. Each mesh is managed by a separate control plane. More information and examples can be found [here](https://github.com/istio-ecosystem/sail-operator/blob/main/docs/README.md#multiple-meshe ...
```
Ok

dhawton commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+* `IstioRevision`: it represents a revision of the control plane.
+* `IstioRevisionTag`: resource represents a stable Revision Tag, which functions as an alias for Istio control plane revisions.
+* `IstioCNI`: Istio's CNI plugin resource.
+* `ZTunnel`: ambient mode Ztunnel DaemonSet (alpha feature).
```
```suggestion
* `ZTunnel`: ambient mode ztunnel `DaemonSet` (alpha feature).
```

dhawton commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+The main resources that are part of the Sail operator are:
+* `Istio`: manages your istio control plane.
+* `IstioRevision`: it represents a revision of the control plane.
+* `IstioRevisionTag`: resource represents a stable Revision Tag, which functions as an alias for Istio control plane revisions.
```
```suggestion
* `IstioRevisionTag`: resource represents a stable revision tag, which functions as an alias for Istio control plane revisions.
```

dhawton commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+* `IstioCNI`: Istio's CNI plugin resource.
+* `ZTunnel`: ambient mode Ztunnel DaemonSet (alpha feature).
+
+Note that if you are migrating from the (now deprecated and deleted) In-Cluster Istio operator, you can check this section in our [documentation](https://github.com/istio-ecosystem/sail-operator/tree/main/docs#migrating-from-istio-in-cluster-operator) where we explain the equivalence of resources, or you can also try our [resource converter](https://github.com/istio-ecosystem/sail-operator/tree/m ...
```
```suggestion
Note that if you are migrating from the (now deprecated and deleted) In-Cluster Istio operator, you can check this section in our [documentation](https://github.com/istio-ecosystem/sail-operator/tree/main/docs#migrating-from-istio-in-cluster-operator) where we explain the equivalence of resources, or you can also try our [resource converter](https://github.com/istio-ecosystem/sail-operator/tree/main/docs#converter-script) to easily convert your `IstioOperator` resource to an `Istio` resource.
```

dhawton commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+{{< /text >}}
+
+The Sail Operator automatically detects whether a given Istio control plane is being used and writes this information in the "In Use" status condition that you see above. Right now, all `IstioRevisions` and our `IstioRevisionTag` are considered "In Use":
+* The old revision `default-v1-24-2` is considered in use because it is referenced by the sample application’s sidecar
```
```suggestion
* The old revision `default-v1-24-2` is considered in use because it is referenced by the sample application’s sidecar.
```

Consistency.

dhawton reviewed (changes requested):
Initial pass of just the markdown, will review formatting on next sweep. Need @istio/technical-oversight-committee approval before docs can approve.

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+---
+title: "Sail operator 1.0.0 has been released: manage Istio with an operator"
```
per https://istio.io/latest/blog/2024/introducing-sail-operator/ and https://github.com/istio-ecosystem/sail-operator, it's "Sail Operator" throughout

craigbox replied:
(Please make sure Istio is capitalized where appropriate too, in case I missed any)

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+---
+title: "Sail operator 1.0.0 has been released: manage Istio with an operator"
```
```suggestion
title: "Sail Operator 1.0.0 released: manage Istio with an operator"
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+---
+title: "Sail operator 1.0.0 has been released: manage Istio with an operator"
+description: Sail Operator 1.0.0 released, a project part of the istio-ecosystem organization. Let's dive in into the basics of the Sail Operator and let's run an example to show how easy it is to manage Istio with it.
```
see https://deploy-preview-16332--preliminary-istio.netlify.app/latest/blog/2025/sail-operator-ga/ - this is too long

```suggestion
description: Dive in into the basics of the Sail Operator and check out an example to see how easy it is to use it to manage Istio.
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+keywords: [istio,operator,sail,incluster,istiooperator]
+---
+
+The Sail operator is GA with a clear mission: to simplify and streamline Istio management in your cluster. By using the Kubernetes operator pattern, Sail operator aims to add new value to istio by improving the upgrade process.
```
Probably want to start with an overview/recap of what the Sail Operator is and a link back to https://istio.io/latest/blog/2024/introducing-sail-operator/

```suggestion
The [Sail Operator](https://github.com/istio-ecosystem/sail-operator) is a community project launched by Red Hat to build a modern [operator](https://www.redhat.com/en/topics/containers/what-is-a-kubernetes-operator) for Istio. [First announced in August 2024](https://github.com/istio-ecosystem/sail-operator) we are pleased to announce Sail Operator is now GA with a clear mission: to simplify and streamline Istio management in your cluster..
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+
+## Simplified Deployment & Management
+
+The Sail operator is engineered to cut down the complexity of installing and running Istio. It automates manual tasks, ensuring a consistent, reliable, and uncomplicated experience from initial installation to ongoing maintenance and upgrades of Istio versions in your cluster. Besides this, the Sail operator APIs are built around Istio’s Helm chart APIs, which means that all the istio configurati ...
```
```suggestion
The Sail Operator is engineered to cut down the complexity of installing and running Istio. It automates manual tasks, ensuring a consistent, reliable, and uncomplicated experience from initial installation to ongoing maintenance and upgrades of Istio versions in your cluster. The Sail Operator APIs are built around Istio’s Helm chart APIs, which means that all the istio configurations are available through the Sail Operator CRD’s values.
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+
+The Sail operator is engineered to cut down the complexity of installing and running Istio. It automates manual tasks, ensuring a consistent, reliable, and uncomplicated experience from initial installation to ongoing maintenance and upgrades of Istio versions in your cluster. Besides this, the Sail operator APIs are built around Istio’s Helm chart APIs, which means that all the istio configurati ...
+
+We encourage users to go through our live [documentation](https://github.com/istio-ecosystem/sail-operator/tree/main/docs) to learn more about this new way to manage your istio control plane.
```
```suggestion
We encourage users to go through our [documentation](https://github.com/istio-ecosystem/sail-operator/tree/main/docs) to learn more about this new way to manage your Istio environment.
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+
+We encourage users to go through our live [documentation](https://github.com/istio-ecosystem/sail-operator/tree/main/docs) to learn more about this new way to manage your istio control plane.
+
+The main resources that are part of the Sail operator are:
```
```suggestion
The main resources that are part of the Sail Operator are:
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+We encourage users to go through our live [documentation](https://github.com/istio-ecosystem/sail-operator/tree/main/docs) to learn more about this new way to manage your istio control plane.
+
+The main resources that are part of the Sail operator are:
+* `Istio`: manages your istio control plane.
+* `IstioRevision`: it represents a revision of the control plane.
+* `IstioRevisionTag`: resource represents a stable revision tag, which functions as an alias for Istio control plane revisions.
+* `IstioCNI`: Istio's CNI plugin resource.
+* `ZTunnel`: ambient mode ztunnel `DaemonSet` (alpha feature).
```
```suggestion
* `Istio`: manages an Istio control plane.
* `IstioRevision`: represents a revision of the control plane.
* `IstioRevisionTag`: represents a stable revision tag, which functions as an alias for an Istio control plane revision.
* `IstioCNI`: manages Istio's CNI node agent.
* `ZTunnel`: manage the ambient mode ztunnel DaemonSet (Alpha feature).
```

confirm this is ZTunnel with a capital Z and T?  We don't usually spell it that way.

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+* `IstioCNI`: Istio's CNI plugin resource.
+* `ZTunnel`: ambient mode ztunnel `DaemonSet` (alpha feature).
+
+Note that if you are migrating from the (now deprecated and deleted) In-Cluster Istio operator, you can check this section in our [documentation](https://github.com/istio-ecosystem/sail-operator/tree/main/docs#migrating-from-istio-in-cluster-operator) where we explain the equivalence of resources, or you can also try our [resource converter](https://github.com/istio-ecosystem/sail-operator/tree/m ...
```
```suggestion
{{< idea >}}
If you are migrating from the [since-removed Istio in-cluster operator](https://istio.io/latest/blog/2024/in-cluster-operator-deprecation-announcement/), you can check this section in our [documentation](https://github.com/istio-ecosystem/sail-operator/tree/main/docs#migrating-from-istio-in-cluster-operator) where we explain the equivalence of resources, or you can also try our [resource converter](https://github.com/istio-ecosystem/sail-operator/tree/main/docs#converter-script) to easily convert your `IstioOperator` resource to an `Istio` resource.
{{< /idea >}}
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+
+Note that if you are migrating from the (now deprecated and deleted) In-Cluster Istio operator, you can check this section in our [documentation](https://github.com/istio-ecosystem/sail-operator/tree/main/docs#migrating-from-istio-in-cluster-operator) where we explain the equivalence of resources, or you can also try our [resource converter](https://github.com/istio-ecosystem/sail-operator/tree/m ...
+
+## Main Features and support
```
```suggestion
## Main features and support
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+## Main Features and support
+
+- Each component of the Istio control plane is managed independently by the Sail Operator through dedicated Kubernetes Custom Resources (CRs). The Sail Operator provides separate CRDs for components such as `Istio`, `IstioCNI`, and `ZTunnel`, allowing you to configure, manage, and upgrade them individually. Additionally, there are CRDs for `IstioRevision` and `IstioRevisionTag` to manage Istio co ...
+- Support for multiple Istio versions. Currently the 1.0.0 version supports: 1.24.3, 1.24.2, 1.24.1, 1.23.5, 1.23.4, 1.23.3, 1.23.0.
```
not 1.25?

fjglira replied:
Hi, when we set everything for `1.0.0` version, `v1.25` was not out yet. So, will be added in a next release

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+- Support for multiple Istio versions. Currently the 1.0.0 version supports: 1.24.3, 1.24.2, 1.24.1, 1.23.5, 1.23.4, 1.23.3, 1.23.0.
+- Two update strategies are supported: `InPlace` and `RevisionBased`. Check our documentation for more information about the update types supported.
+- Support for multicluster Istio [deployment models](/docs/setup/install/multicluster/): multi-primary, primary-remote, external control plane. More information and examples in our [documentation](https://github.com/istio-ecosystem/sail-operator/blob/main/docs/README.md#multi-cluster).
+- Ambient is alpha: check our specific [documentation](https://github.com/istio-ecosystem/sail-operator/blob/main/docs/common/istio-ambient-mode.md).
```
```suggestion
- Ambient mode support is Alpha: check our specific [documentation](https://github.com/istio-ecosystem/sail-operator/blob/main/docs/common/istio-ambient-mode.md).
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+- Two update strategies are supported: `InPlace` and `RevisionBased`. Check our documentation for more information about the update types supported.
+- Support for multicluster Istio [deployment models](/docs/setup/install/multicluster/): multi-primary, primary-remote, external control plane. More information and examples in our [documentation](https://github.com/istio-ecosystem/sail-operator/blob/main/docs/README.md#multi-cluster).
+- Ambient is alpha: check our specific [documentation](https://github.com/istio-ecosystem/sail-operator/blob/main/docs/common/istio-ambient-mode.md).
+- Addons are managed separately from the Sail Operator. They can be easily integrated with the Sail operator, you can check this section for the [documentation](https://github.com/istio-ecosystem/sail-operator/blob/main/docs/README.md#addons) for examples and more information.
```
```suggestion
- Addons are managed separately from the Sail Operator. They can be easily integrated with the Sail Operator, check this section for the [documentation](https://github.com/istio-ecosystem/sail-operator/blob/main/docs/README.md#addons) for examples and more information.
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+
+## Why Now?
+
+As cloud-native architectures continue to evolve, a robust and user-friendly operator for Istio is more essential than ever. The Sail Operator offers developers and operations teams a consistent, secure, and efficient solution that feels familiar to those used to working with operators. Its GA release signals a mature solution, ready to support even the most demanding production environments.
```
```suggestion
As cloud native architectures continue to evolve, we feel a robust and user-friendly operator for Istio is more essential than ever. The Sail Operator offers developers and operations teams a consistent, secure, and efficient solution that feels familiar to those used to working with operators. Its GA release signals a mature solution, ready to support even the most demanding production environments.
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+- Ambient is alpha: check our specific [documentation](https://github.com/istio-ecosystem/sail-operator/blob/main/docs/common/istio-ambient-mode.md).
+- Addons are managed separately from the Sail Operator. They can be easily integrated with the Sail operator, you can check this section for the [documentation](https://github.com/istio-ecosystem/sail-operator/blob/main/docs/README.md#addons) for examples and more information.
+
+## Why Now?
```
```suggestion
## Why now?
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+
+The Sail operator is GA with a clear mission: to simplify and streamline Istio management in your cluster. By using the Kubernetes operator pattern, Sail operator aims to add new value to istio by improving the upgrade process.
+
+## Simplified Deployment & Management
```
```suggestion
## Simplified deployment & management
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+
+## Try it out
+
+Do you want to try our operator?
+Going over this example will show you how to safely do an update of your istio control plane by using the revision-based upgrade strategy, this means you will have two Istio control planes running at the same time, allowing you to migrate workloads easily, minimizing the risk of traffic disruptions.
```
```suggestion
Would you like to try out Sail Operator?
This example will show you how to safely do an update of your Istio control plane by using the revision-based upgrade strategy. This means you will have two Istio control planes running at the same time, allowing you to migrate workloads easily, minimizing the risk of traffic disruptions.
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+- Kubectl
+- Istioctl
+
+### Install the operator from the sail operator helm repository
```
```suggestion
### Install the operator from the Sail Operator Helm repository
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+sail-operator-56bf994f49-j67ft   1/1     Running   0          87s
+{{< /text >}}
+
+#### Create an Istio resource with istio version 1.24.2 and a Istio revision tag
```
```suggestion
#### Create an Istio resource with Istio v1.24.2 and a revision tag
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+default   NotReferencedByAnything   False    default-v1-24-2   4m43s
+{{< /text >}}
+
+Note that the `IstioRevisionTag` status is `NotReferencedByAnything`, this is because there are currently no resources using the revision `default-v1-24-2`.
```
```suggestion
Note that the `IstioRevisionTag` status is `NotReferencedByAnything`. This is because there are currently no resources using the revision `default-v1-24-2`.
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+
+### Deploy sample application
+
+- Create a namespace and label it to enable istio injection
```
```suggestion
- Create a namespace and label it to enable Istio injection:
```

craigbox replied:
```suggestion
Create a namespace and label it to enable istio injection
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+$ kubectl label namespace sample istio-injection=enabled
+{{< /text >}}
+
+After labeling the namespace you will see that the `IstioRevisionTag` resource status will change to In Use true, this is because there is now a resource using the revision `default-v1-24-2`.
```
```suggestion
After labeling the namespace you will see that the `IstioRevisionTag` resource status will change to 'In Use: True', because there is now a resource using the revision `default-v1-24-2`.
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+$ kubectl get istiorevisiontag
+{{< /text >}}
+
+{{< text plain >}}
+NAME      STATUS    IN USE   REVISION          AGE
+default   Healthy   True     default-v1-24-2   6m24s
+{{< /text >}}
```
```suggestion
    {{< text plain >}}
    NAME      STATUS    IN USE   REVISION          AGE
    default   Healthy   True     default-v1-24-2   6m24s
    {{< /text >}}
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+
+After labeling the namespace you will see that the `IstioRevisionTag` resource status will change to In Use true, this is because there is now a resource using the revision `default-v1-24-2`.
+
+{{< text bash >}}
+$ kubectl get istiorevisiontag
+{{< /text >}}
```
```suggestion
    {{< text bash >}}
    $ kubectl get istiorevisiontag
    {{< /text >}}
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+Check the state of the resources created:
+- `istiod` pods are running
+
+{{< text bash >}}
+$ kubectl get pods -n istio-system
+{{< /text >}}
+
+{{< text plain >}}
+NAME                                    READY   STATUS    RESTARTS   AGE
+istiod-default-v1-24-2-bd8458c4-jl8zm   1/1     Running   0          3m45s
+{{< /text >}}
```
```suggestion
    {{< text bash >}}
    $ kubectl get pods -n istio-system
    {{< /text >}}

    {{< text plain >}}
    NAME                                    READY   STATUS    RESTARTS   AGE
    istiod-default-v1-24-2-bd8458c4-jl8zm   1/1     Running   0          3m45s
    {{< /text >}}
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+
+- `Istio` resource created
+
+{{< text bash >}}
+$ kubectl get istio
+{{< /text >}}
```
```suggestion
    {{< text bash >}}
    $ kubectl get istio
    {{< /text >}}
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+$ kubectl get istio
+{{< /text >}}
+
+{{< text plain >}}
+NAME      REVISIONS   READY   IN USE   ACTIVE REVISION   STATUS    VERSION   AGE
+default   1           1       1        default-v1-24-2   Healthy   v1.24.2   4m27s
+{{< /text >}}
```
```suggestion
    {{< text plain >}}
    NAME      REVISIONS   READY   IN USE   ACTIVE REVISION   STATUS    VERSION   AGE
    default   1           1       1        default-v1-24-2   Healthy   v1.24.2   4m27s
    {{< /text >}}
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+
+- `IstioRevisionTag` resource created
+
+{{< text bash >}}
+$ kubectl get istiorevisiontag
+{{< /text >}}
+
+{{< text plain >}}
+NAME      STATUS                    IN USE   REVISION          AGE
+default   NotReferencedByAnything   False    default-v1-24-2   4m43s
+{{< /text >}}
```
```suggestion
    {{< text bash >}}
    $ kubectl get istiorevisiontag
    {{< /text >}}

    {{< text plain >}}
    NAME      STATUS                    IN USE   REVISION          AGE
    default   NotReferencedByAnything   False    default-v1-24-2   4m43s
    {{< /text >}}
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+default   Healthy   True     default-v1-24-2   6m24s
+{{< /text >}}
+
+- Deploy the sample application
```
```suggestion
Deploy the sample application:
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+$ kubectl apply -f {{< github_file >}}/samples/sleep/sleep.yaml -n sample
+{{< /text >}}
+
+- Confirm istio proxy version of the sample app match the control plane version
```
```suggestion
Confirm the proxy version of the sample app matches the control plane version:
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+
+#### Upgrade the Istio control plane to version 1.24.3
+
+- Update the `Istio` resource with the new version
```
```suggestion
Update the `Istio` resource with the new version:
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+$ kubectl patch istio default -n istio-system --type='merge' -p '{"spec":{"version":"v1.24.3"}}'
+{{< /text >}}
+
+- Check the `Istio` resource, you will see that there are two revisions and there are two `Istio` ready
```
```suggestion
Check the `Istio` resource. You will see that there are two revisions and they are both 'ready':
```

if this is correct

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+default   2           2       2        default-v1-24-3   Healthy   v1.24.3   10m
+{{< /text >}}
+
+- The `IstioRevisiontag` now references the new revision
```
```suggestion
The `IstioRevisiontag` now references the new revision:
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+default   Healthy   True     default-v1-24-3   11m
+{{< /text >}}
+
+- There are two `IstioRevisions`, one for each istio version
```
```suggestion
There are two `IstioRevisions`, one for each Istio version:
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+* The new revision `default-v1-24-3` is considered in use because it is referenced by the tag.
+* The tag is considered in use because it is referenced by the sample namespace.
+
+- Confirm there are two control plane pods running, one for each revision
```
```suggestion
Confirm there are two control plane pods running, one for each revision:
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+istiod-default-v1-24-3-68df97dfbb-v7ndm   1/1     Running   0          6m32s
+{{< /text >}}
+
+- Confirm the proxy sidecar version remains the same
```
```suggestion
Confirm the proxy sidecar version remains the same:
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+sleep-5fcd8fd6c8-q4c9x.sample     Kubernetes     SYNCED (6m40s)     SYNCED (6m40s)     SYNCED (6m40s)     SYNCED (6m40s)     IGNORED     istiod-default-v1-24-2-bd8458c4-jl8zm     1.24.2
+{{< /text >}}
+
+- Restart the sample pod
```
```suggestion
Restart the sample pod:
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+$ kubectl rollout restart deployment -n sample
+{{< /text >}}
+
+- Confirm the proxy sidecar version is updated
```
```suggestion
Confirm the proxy sidecar version is updated:
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+sleep-6f87fcf556-k9nh9.sample     Kubernetes     SYNCED (29s)     SYNCED (29s)     SYNCED (29s)     SYNCED (29s)     IGNORED     istiod-default-v1-24-3-68df97dfbb-v7ndm     1.24.3
+{{< /text >}}
+
+- When an `IstioRevision` is no longer in use and is not the active revision of an `Istio` resource (for example, when it is not the version that is set in the `spec.version` field), the Sail Operator will delete it after a grace period, which defaults to 30 seconds. Confirm the deletion of the old control plane and `IstioRevision`:
```
```suggestion
When an `IstioRevision` is no longer in use and is not the active revision of an `Istio` resource (for example, when it is not the version that is set in the `spec.version` field), the Sail Operator will delete it after a grace period, which defaults to 30 seconds. Confirm the deletion of the old control plane and `IstioRevision`:
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+
+## Conclusion
+
+The Sail Operator automates manual tasks, ensuring a consistent, reliable, and uncomplicated experience from initial installation to ongoing maintenance and upgrades of Istio in your cluster. The Sail Operator is an istio-ecosystem project, and we encourage you to try it out and provide feedback to help us improve it, you can check our contribution guide [here](https://github.com/istio-ecosystem/ ...
```
```suggestion
The Sail Operator automates manual tasks, ensuring a consistent, reliable, and uncomplicated experience from initial installation to ongoing maintenance and upgrades of Istio in your cluster. The Sail Operator is an [istio-ecosystem ](https://github.com/istio-ecosystem)project, and we encourage you to try it out and provide feedback to help us improve it, you can check our [contribution guide](https://github.com/istio-ecosystem/sail-operator/blob/main/CONTRIBUTING.md) for more information about how to contribute to the project.
```

craigbox commented on content/en/blog/2025/sail-operator-ga/index.md:
```diff
@@ -0,0 +1,301 @@
+
+As cloud-native architectures continue to evolve, a robust and user-friendly operator for Istio is more essential than ever. The Sail Operator offers developers and operations teams a consistent, secure, and efficient solution that feels familiar to those used to working with operators. Its GA release signals a mature solution, ready to support even the most demanding production environments.
+
+## Try it out
```
Example content in blogs ages poorly for things which can end up having high SEO.  Have you considered having this live somewhere in your docs and just be pointed to by here?

fjglira replied:
Yes, it can be a good way to keep the example up-to-date, but I was trying to give an example to get the attention of the users reading the blog post directly. Let me check the best way to move this example to the docs

fjglira replied:
I changed the entire `Try it out` section pointing to our example in the docs

craigbox replied:
TBH (and with my apologies), what is left is half way. Right now you're only demonstrating installing, which is just a Helm command, and shows nothing about how easy Sail Operator is to use.  I would either add it back, or remove the code entirely, in favour of the "Do you want to.." and below.

fjglira replied:
I'll add it back, the example will always work for the specific version `1.0.0`. Maybe add a note to visit the Sail page to check the latest version and updated documentation in case to want to try the latest features

fjglira replied:
Hi, I added the example back and also a Note to point to the latest release version as an example in case this post has traffic when a new version is out

istio-testing commented:
@fjglira: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
lint_istio.io | 25ef08fad1ca4a1cffd296ab268ba9b29add326e | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/16332/lint_istio.io/1907013719926247424) | true | `/test lint`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@fjglira: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
lint_istio.io | 429387c8f40e9699256e327dcec1374bdca0c06c | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/16332/lint_istio.io/1907020923823722496) | true | `/test lint`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@fjglira: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
lint_istio.io | 6cf72598b8a41c97b927b2866df780eb7316c193 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/16332/lint_istio.io/1907066638843252736) | true | `/test lint`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

fjglira reviewed (commented):
Hi @craigbox, I just make simpler the Try out section and pointed to the example in the docs that address the same example that I was describing in the blogpost

istio-testing commented:
@fjglira: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
lint_istio.io | 8bf0e35a274846bc5d275a51fc6f8db7bb0d0c0f | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/16332/lint_istio.io/1907375898605129728) | true | `/test lint`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

craigbox commented:
I've changed all the `{{ text bash }} {{ text plain }}` blocks to be just `{{ text bash }}`, as it renders nicer, and the "copy code" button already works to only take the first command and remove the $.

I've also updated the publish date to today and fixed another couple of nits, so when I get the lint right I will approve and publish for you.

craigbox reviewed (approved):
/cherry-pick release-1.25

fjglira commented:
> I've changed all the `{{ text bash }} {{ text plain }}` blocks to be just `{{ text bash }}`, as it renders nicer, and the "copy code" button already works to only take the first command and remove the $.
> 
> I've also updated the publish date to today and fixed another couple of nits, so when I get the lint right I will approve and publish for you.

Thanks a lot for all the help @craigbox

fjglira commented:
/cherry-pick release-1.25

istio-testing commented:
@fjglira: only [istio](https://github.com/orgs/istio/people) org members may request cherry picks. If you are already part of the org, make sure to [change](https://github.com/orgs/istio/people?query=fjglira) your membership to public. Otherwise you can still do the cherry-pick manually. 

<details>

In response to [this](https://github.com/istio/istio.io/pull/16332#issuecomment-2777748333):

>/cherry-pick release-1.25


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository.
</details>

fjglira commented:
Hi @craigbox, for some reason the cherry-pick was not created from your `/cherry-pick release-1.25` comment

craigbox commented:
ok, not sure what is happening sorry @fjglira.  Feel free to manually CP if it doesn't resolve in a few hours.

fjglira commented:
> ok, not sure what is happening sorry @fjglira. Feel free to manually CP if it doesn't resolve in a few hours.

Sure, I'll do it later today if still is not solved
